engine/src/main/java/org/pentaho/di/trans/steps/mappinginput/MappingInput.java

```diff
@@ -0,0 +1,229 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.mappinginput;
+
+import java.util.List;
+
+import org.pentaho.di.core.BlockingRowSet;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleStepException;
+import org.pentaho.di.core.row.RowDataUtil;
+import org.pentaho.di.core.row.RowMeta;
+import org.pentaho.di.core.row.ValueMetaInterface;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.trans.Trans;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.BaseStep;
+import org.pentaho.di.trans.step.StepDataInterface;
+import org.pentaho.di.trans.step.StepInterface;
+import org.pentaho.di.trans.step.StepMeta;
+import org.pentaho.di.trans.step.StepMetaInterface;
+import org.pentaho.di.trans.steps.mapping.MappingValueRename;
+
+/**
+ * Do nothing. Pass all input data to the next steps.
+ *
+ * @author Matt
+ * @since 2-jun-2003
+ */
+public class MappingInput extends BaseStep implements StepInterface {
+
+  private static Class<?> PKG = MappingInputMeta.class; // for i18n purposes, needed by Translator2!!
+  private int timeOut = 60000;
+  private MappingInputMeta meta;
+
+  private MappingInputData data;
+
+  public MappingInput( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr, TransMeta transMeta,
+      Trans trans ) {
+    super( stepMeta, stepDataInterface, copyNr, transMeta, trans );
+  }
+
+  public void setTimeOut( int timeOut ) {
+    this.timeOut = timeOut;
+  }
+
+  // ProcessRow is not doing anything
+  // It's a place holder for accepting rows from the parent transformation...
+  // So, basically, this is a glorified Dummy with a little bit of meta-data
+  //
+  public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
+    meta = (MappingInputMeta) smi;
+    data = (MappingInputData) sdi;
+
+    if ( !data.linked ) {
+      //
+      // Wait until we know were to read from the parent transformation...
+      // However, don't wait forever, if we don't have a connection after 60 seconds: bail out!
+      //
+      int totalsleep = 0;
+      while ( !isStopped() && data.sourceSteps == null ) {
+        try {
+          totalsleep += 10;
+          Thread.sleep( 10 );
+        } catch ( InterruptedException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)